### PR TITLE
[RUNTIME] add ability to terminate/pause/resume a wasm program

### PR DESCRIPTION
**Pull request by Officeyutong**
I write a new function `run_wasm_bpf_module_async` to provide the ability to run a wasm program async and to control it.

- It has the same signature as `run_wasm_bpf_module`.
- This function will return immediately after calling, and leave the wasm program running in another thread
- It will return two handles. `WasmProgramHandle` provides ability to `pause`/`resume`/`terminate` a wasm program. `JoinHandle` provides the ability to wait for the finish of the wasm program. 
- I use the epoch interruption and the corresponding callback function (which will block the execution of the wasm program) to implement the functions. See https://docs.rs/wasmtime/latest/wasmtime/struct.Store.html#method.epoch_deadline_callback for details.
- I use mpsc channels to implement inter-thread communication. For example, send `ProgramOperation::Resume` signal to a hanging thread which is stuck on receiving the next operation.

### CODE REVIEW

**Review by yunwei37 — CHANGES_REQUESTED** [review]

**Comment by yunwei37** [comment]
nice job!

File: runtime/wasm-bpf-rs/tests/tick/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+tick.wasm
```
[the remark this answers is not in the document]
Comment (Officeyutong) [reply runtime/wasm-bpf-rs/tests/tick/.gitignore:1]: My  purpose is to ignore the tick.wasm under tick directory

File: runtime/wasm-bpf-rs/src/lib.rs [hunk 2]
```diff
@@ -36,8 +34,7 @@ pub struct Config {
     pub stdout: Box<dyn WasiFile>,
     /// stderr file for sending error to the host
     pub stderr: Box<dyn WasiFile>,
-    /// Whether enable epoch interruption
-    pub enable_epoch_interruption: bool,
+    // Now we force epoch interruption to be enabled
```
[the remark this answers is not in the document]
Comment (Officeyutong) [reply runtime/wasm-bpf-rs/src/lib.rs]: Fixed

File: runtime/wasm-bpf-rs/src/handle.rs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+use std::sync::mpsc;
+
+use anyhow::{anyhow, bail, Context};
+use wasmtime::Engine;
+
+pub enum ProgramOperation {
```
[the remark this answers is not in the document]
Comment (Officeyutong) [reply runtime/wasm-bpf-rs/src/handle.rs:7]: Done

**Review by yunwei37 — APPROVED** [review]

**Pull request merged**